### PR TITLE
Add outputs for peer-reviewed step and restructured steps

### PR DESCRIPTION
**Pull request by HazalCiplak**
https://github.com/elifesciences/data-hub-issues/issues/574

### CODE REVIEW

File: tests/unit_tests/docmaps/provider_test.py [hunk 1]
```diff
@@ -232,18 +211,42 @@ def test_should_populate_second_step_actions_outputs_with_necessary_details(self
...
-        first_step = docmaps_item['steps'][first_step_key]
-        assert first_step['actions'] == [
+        peer_reviewed_step = docmaps_item['steps']['_:b2']
+        assert peer_reviewed_step['actions'] == [
             {
                 'participants': [],
                 'outputs': [
                     {
-                        'type': '',
+                        'type': 'review-article',
```
Comment (de-code) [inline tests/unit_tests/docmaps/provider_test.py]: just reading the test it is a bit difficult to see why we are for example expecting this type to be a `review-article`.

File: data_hub_api/docmaps/provider.py [hunk 2]
```diff
@@ -84,87 +88,90 @@ def get_docmap_assertions_value_for_preprint_under_review_step(
...
-            'type': 'preprint',
-            'doi': elife_doi,
-            'url': elife_doi_url,
-            'content': [{
-                'type': 'web-page',
-                'url': f'{ELIFE_REVIEWED_PREPRINTS_URL}{manuscript_id}'
-            }]
-        }]
-    }
+def get_outputs_type_for_peer_reviewed_step(
```
Comment (de-code) [inline data_hub_api/docmaps/provider.py]: I think it would be good to have a separate test for this function to test with a few tag combinations.
The default should probably be `None`.
Only if we found a related tag, we should recognise it as a `review-article`.

File: data_hub_api/docmaps/provider.py [hunk 3]
```diff
@@ -194,85 +201,77 @@ def get_single_actions_value_for_preprint_peer_reviewed_step(
...
+        elif step_number == 2:  # peer-reviewed
+            step_dict = get_docmaps_step_for_peer_reviewed_status(
+                query_result_item=query_result_item
+            )
+        steps_dict['_:b'+str(step_number)] = dict(step_dict, **step_ranking_dict)
         step_number += 1
     return remove_key_with_none_value_only(steps_dict)
 
 
+def get_docmap_step(query_result_item: dict) -> int:
```
Comment (de-code) [inline data_hub_api/docmaps/provider.py]: This should probably be `get_docmap_step_count` or `get_docmap_step_count_for_query_result_item`?

Comment (de-code) [reply data_hub_api/docmaps/provider.py]: I still believe that tying it to the count of steps makes it a less readable as the step index becomes a magic number.

Instead something simple like this would communicate clearly what steps it generates and under what condition:

```python
def iter_docmap_steps_for_query_result_item(query_result_item: dict) -> Iterable[dict]:
    yield get_manuscript_published_step_for_query_result_item(query_result_item)
    yield get_under_review_step_for_query_result_item(query_result_item)
    if not query_result_item['evaluations']:
        yield get_peer_reviewed_step_for_query_result_item(query_result_item)
```

File: data_hub_api/docmaps/provider.py (hunk 2 above)
Comment (HazalCiplak) [reply data_hub_api/docmaps/provider.py]: If there is unrelated tags probably we need to filter out these data for peer-reviewed step. Because according to examples it's only expecting two types (`review-article` and `evaluation-summary`) and we don't need  `None` types. What do you think?
And what should be the related tag for `review-article`?

Comment (de-code) [reply data_hub_api/docmaps/provider.py]: Yes, `None` would signify that we don't want to include them alternatively we could raise an error. But not including them seems okay in this case.
(And there is the question whether we want to include author responses in the reviewed-preprint step or not)

File: tests/unit_tests/docmaps/provider_test.py [hunk 4]
```diff
@@ -333,7 +383,7 @@ def test_should_add_has_evaluatons_where_clause_to_query(
             only_include_reviewed_preprint_type=False,
             only_include_evaluated_preprints=True
         )
-        assert provider.docmaps_index_query.rstrip().endswith('WHERE has_evaluations')
+        assert provider.docmaps_index_query.rstrip().endswith('WHERE has_evaluations\nLIMIT 20')
```
Comment (de-code) [inline tests/unit_tests/docmaps/provider_test.py:430]: this seems an unrelated change. (maybe we could just add a `max_rows` API parameter?)

File: data_hub_api/docmaps/provider.py (hunk 3 above)
Comment (HazalCiplak) [reply data_hub_api/docmaps/provider.py]: created seperate issue for this
